tests/ops/arithmetic/test_adjoint_op.py

```diff
@@ -0,0 +1,437 @@
+        assert qml.math.allclose(op.data, [2.0, 1.0])
+
+        assert op.wires == qml.wires.Wires([0, "b"])
+
+
+class TestProperties:
+    """Test Adjoint properties."""
+
+    def test_data(self):
+        """Test base data can be get and set through Adjoint class."""
+        x = np.array(1.234)
+
+        base = qml.RX(x, wires="a")
+        adj = Adjoint(base)
+
+        assert adj.data == [x]
+
+        # update parameters through adjoint
+        x_new = np.array(2.3456)
+        adj.data = [x_new]
+        assert base.data == [x_new]
+        assert adj.data == [x_new]
+
+        # update base data updates Adjoint data
+        x_new2 = np.array(3.456)
+        base.data = [x_new2]
+        assert adj.data == [x_new2]
+
+    def test_has_matrix_true(self):
+        """Test `has_matrix` property carries over when base op defines matrix."""
+        base = qml.PauliX(0)
+        op = Adjoint(base)
+
+        assert op.has_matrix
+
+    def test_has_matrix_false(self):
+        """Test has_matrix property carries over when base op does not define a matrix."""
+        base = qml.QubitStateVector([1, 0], wires=0)
+        op = Adjoint(base)
+
+        assert not op.has_matrix
+
+    def test_queue_category(self):
+        """Test that the queue category `"_ops"` carries over."""
+        op = Adjoint(qml.PauliX(0))
+        assert op._queue_category == "_ops"
+
+    def test_queue_category_None(self):
+        """Test that the queue category `None` for some observables carries over."""
+        op = Adjoint(qml.PauliX(0) @ qml.PauliY(1))
+        assert op._queue_category is None
+
+
+class TestMiscMethods:
+    """Test miscellaneous small methods on the Adjoint class."""
+
+    def test_label(self):
+        """Test that the label method for the adjoint class adds a † to the end."""
+        base = qml.Rot(1.2345, 2.3456, 3.4567, wires="b")
+        op = Adjoint(base)
+        assert op.label(decimals=2) == "Rot\n(1.23,\n2.35,\n3.46)†"
+
+    def test_adjoint_of_adjoint(self):
+        """Test that the adjoint of an adjoint is the original operation."""
+        base = qml.PauliX(0)
+        op = Adjoint(base)
+
+        assert op.adjoint() is base
+
+    def test_diagonalizing_gates(self):
+        """Assert that the diagonalizing gates method gives the base's diagonalizing gates."""
+        base = qml.Hadamard(0)
+        diag_gate = Adjoint(base).diagonalizing_gates()[0]
+
+        assert isinstance(diag_gate, qml.RY)
+        assert qml.math.allclose(diag_gate.data[0], -np.pi / 4)
+
+
+class TestOperationSpecificMethProp:
+    def test_generator(self):
+        """Assert that the generator of an Adjoint is -1.0 times the base generator."""
+        base = qml.RX(1.23, wires=0)
+        op = Adjoint(base)
+
+        assert base.generator().compare(-1.0 * op.generator())
+
+    def test_no_generator(self):
+        """Test that an adjointed non-Operation raises a GeneratorUndefinedError."""
```

Comment:
```suggestion
        """Test that an adjointed non-Operator raises a GeneratorUndefinedError."""
```

What is a non-Operation/Operator?